torch/csrc/autograd/variable.h

```diff
@@ -256,6 +263,35 @@ struct TORCH_API AutogradMeta : public c10::AutogradMetaInterface {
   }
 };
 
+struct TORCH_API ViewInfo {
+  /// The base `Variable`
+  /// This cannot be a view of the same type as this one.
+  Variable base_;
+
+  /// By default we use as_strided to recover views which is more efficient.
+  /// view_fn is only saved when as_strided is not supported.
+  /// If view_fn has value, we use it to recover views in backward.
+  c10::optional<std::function<Variable(const Variable&)>> view_fn_;
```

Comment:
std::function has an empty state. You don't need to wrap it in an optional.

Reply:
I just kept the current type for this and for it's accesor functions. Should I open an issue to fix that?

Reply:
Don't ask to file an issue ;)

Reply:
@swolchok It looks like ViewInfo is getting moved around; I vaguely remember moves on `std::function` being famously slow but I can't find any reference, do you know what I'm thinking of?

Reply:
Nope, I have never heard of that. If the optional would help then that would mean moves of an empty std::function are slow, which would be especially surprising.

Reply:
Issue opened. I can try that and see the perf implication after this PR is done.